collagraph/renderers/pyside_renderer.py

```diff
@@ -219,7 +216,12 @@ def set_attribute(self, el: Any, attr: str, value: Any):
 
     def remove_attribute(self, el: Any, attr: str, value: Any):
         """Remove the attribute `attr` from the element `el`."""
-        raise NotImplementedError
+        # Make it possible to delete custom attributes
+        if attribute := getattr(el, attr, None):
```

Comment:
What if the value is Falsey?

Reply:
🤦‍♂️ yep, not the first time I encounter this gotcha :/

Reply:
I've checked, and removing attributes can be called when widgets/objects are re-used. So we'll need to figure out how to 'remove' non-custom attributes such as `objectName`.

I stumbled upon the `.metaObject` property which allows for getting some info on properties on `QObject`s and for resetting properties. But apparently, not many properties can be reset...
So we might need to resort to keeping track of default values? Or defining a list of default values per type?

Reply:
Resetting to default is indeed an option but that can get tricky too. Not too sure what is best... vue-nodegui doesn't seem to do anything comparable

Reply:
I guess you could put some logic in `create_element` to populate a map of all the default property values per type.

Reply:
Yeah, I did just that, and used `QMetaObject` to get the `QMetaProperty` and use that to get the 'default' value. Then I store the QMetaProperty object together with that value. When removing an attribute, then the QMetaProperty can be used to set the default value for the property on the element.

Note that this is now limited to Qt 'properties', but maybe that's all we need for now?

Reply:
> Under what circumstances are elements re-used?

```
<template>
  <widget>
    <widget
      v-if="foo"
      object_name="foo"
    />
    <widget />
    <button text="toggle" @clicked="toggle" />
  </widget>
</template>
```
When the first widget element is 'removed', then actually the second Qt widget will be destroyed and the first widget will be repurposed to be like the second widget. So in this case, the `object_name` property would have to be removed from the first widget. Does this answer your question?

Reply:
I think not, because you have to pass the object for the read and write operations. So I guess it doesn't have a link to the object. But I will check it just to make sure.

Reply:
```
>>> from PySide6 import QtWidgets
>>> app = QtWidgets.QApplication()
>>> widget = QtWidgets.QWidget()
>>> prop = widget.metaObject().property(0)
>>> prop
<PySide6.QtCore.QMetaProperty object at 0x119ecd8c0>
>>> from weakref import ref
>>> widget_ref = ref(widget)
>>> widget_ref
<weakref at 0x119ecf180; to 'PySide6.QtWidgets.QWidget' at 0x119ecd840>
>>> del widget
>>> widget_ref
<weakref at 0x119ecf180; dead>
>>> prop
<PySide6.QtCore.QMetaProperty object at 0x119ecd8c0>
>>>
```
Works fine!